storage/src/mmr/journaled.rs

```diff
@@ -894,13 +818,23 @@ impl<E: RStorage + Clock + Metrics, D: Digest> DirtyMmr<E, D> {
     }
 }
 
-impl<E: RStorage + Clock + Metrics, D: Digest> Storage<D> for Mmr<E, D, Clean<D>> {
+impl<E: RStorage + Clock + Metrics + Sync, D: Digest, S: State<D> + Send + Sync> Storage<D>
+    for Mmr<E, D, S>
+{
     fn size(&self) -> Position {
         self.size()
     }
 
     async fn get_node(&self, position: Position) -> Result<Option<D>, Error> {
```

Comment:
If the MMR is dirty then there may be a dummy digest at `position`, causing the caller to silently get an invalid value